### PR TITLE
Add Bioconductor translator

### PR DESCRIPTION
**Pull request by hubentu**
The search page ("multiple") worked manually but failed in the testing. Could anyone help to debug? Thanks!

### CODE REVIEW

**Comment by hubentu** [comment]
I have no idea about the error. Could anyone help to check? Thanks!

**Comment by zuphilip** [comment]
This are "only" linting issues which can also be fixed later. It is not an error per se.

However, try to change the line

https://github.com/zotero/translators/blob/229f4aaf14629739f931976bf01325fc03f35f94/Bioconductor.js#L84

into simply  `return;` which jumps out of the function without a return value, which otherwise is not given in all other cases.

**Comment by hubentu** [comment]
@zuphilip Thank you! Finally, all checks have passed. Hope it can be accepted soon :)

**Comment by zuphilip** [comment]
Great! I can't promise to review it fast...

File: Bioconductor.js [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+		}
+	}
+
+	item.versionNumber = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "Version")]/following-sibling::td');
+	item.date = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "Published")]/following-sibling::td');
+	item.rights = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "License")]/following-sibling::td');
+	item.url = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "Package Short Url")]/following-sibling::td') || url;
+	item.company = 'Bioconductor';
+	var year = new Date().getFullYear();
+	item.date = ZU.strToISO(year);
```
[the remark this answers is not in the document]
Comment (hubentu) [reply Bioconductor.js]: You are right. To make it simple, the corresponding release year was extracted to be the date.

File: Bioconductor.js [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+				}
+			}
+		}
+	}
+
+	item.versionNumber = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "Version")]/following-sibling::td');
+	item.date = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "Published")]/following-sibling::td');
+	item.rights = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "License")]/following-sibling::td');
+	item.url = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "Package Short Url")]/following-sibling::td') || url;
+	item.company = 'Bioconductor';
```
[the remark this answers is not in the document]
Comment (hubentu) [reply Bioconductor.js]: I am not sure.. Most of Bioc packages were developed with R and C++.

File: Bioconductor.js [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+	var rows = doc.querySelectorAll('#PageContent > div.do_not_rebase > p');
+	for (let i = 0; i < rows.length; i++) {
+		if (ZU.trimInternal(rows[i].textContent).startsWith('Bioconductor version:')) {
+			item.abstractNote = ZU.trimInternal(rows[i + 1].textContent);
+		}
+		if (ZU.trimInternal(rows[i].textContent).startsWith('Author')) {
+			var authorString = ZU.trimInternal(rows[i].textContent);
+			if (authorString) {
+				var creators = authorString.replace(/Author:\s*/, '').replace(/\[.+?\]/g, '').replace(/\(.+?\)/g, '');
+				creators = creators.split(/,\s*/);
```
[the remark this answers is not in the document]
Comment (hubentu) [reply Bioconductor.js]: Added. Thanks! Packages from old releases might have all kinds of nonstandard items.

File: Bioconductor.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (zuphilip) [reply Bioconductor.js]: Okay, then it is maybe negligable.

File: Bioconductor.js [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+	}
+	else if (detectWeb(doc, url) == "computerProgram") {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var item = new Zotero.Item('computerProgram');
+	item.title = text(doc, '#PageContent > h1') + ": " + text(doc, '#PageContent > div.do_not_rebase > h2');
```
[the remark this answers is not in the document]
Comment (zuphilip) [reply Bioconductor.js]: This one still seems open, i.e. I suggest to do something like
```js
item.title = text(doc, '#PageContent > h1');
var subtitle = text(doc, '#PageContent > div.do_not_rebase > h2');
if (subtitle) {
   item.title += ": " + subtitle;
}
```
(not tested)

File: Bioconductor.js [hunk 5]
```diff
@@ -0,0 +1,341 @@
...
+	}
+	var rows = doc.querySelectorAll('#PageContent > div.do_not_rebase > p');
+	for (let i = 0; i < rows.length; i++) {
+		if (ZU.trimInternal(rows[i].textContent).startsWith('Bioconductor version:')) {
+			item.system = ZU.trimInternal(rows[i].textContent);
+			item.abstractNote = ZU.trimInternal(rows[i + 1].textContent);
+		}
+		if (ZU.trimInternal(rows[i].textContent).startsWith('Author')) {
+			var authorString = ZU.trimInternal(rows[i].textContent);
+			if (authorString) {
```
Comment (zuphilip) [inline Bioconductor.js]: That testing seems to be unnecessary.

File: Bioconductor.js [hunk 6]
```diff
@@ -0,0 +1,341 @@
...
+				creators = creators.split(/,|and\s*/);
+				for (let i = 0; i < creators.length; i++) {
+					item.creators.push(ZU.cleanAuthor(creators[i], 'programmer'));
+				}
+			}
+		}
+	}
+
+	item.versionNumber = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "Version")]/following-sibling::td');
+	item.date = ZU.xpathText(doc, '//table/tbody/tr/td[contains(text(), "Published")]/following-sibling::td');
```
Comment (zuphilip) [inline Bioconductor.js]: This seems to be overwritten some lines below. Should we prefer that date here or is it negligible at all?

File: Bioconductor.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (zuphilip) [reply Bioconductor.js]: Okay. But we have to make this inside a `if (year)` block. Moreover, use either `match` or the `ZU.strToISO` method but not both.

File: Bioconductor.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (hubentu) [reply Bioconductor.js]: Updated. Thanks!

File: Bioconductor.js (hunk 5 above)
Comment (hubentu) [reply Bioconductor.js]: Removed. Thx!

File: Bioconductor.js (hunk 6 above)
Comment (hubentu) [reply Bioconductor.js]: You are right. Removed.

File: Bioconductor.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (hubentu) [reply Bioconductor.js]: The `if` block is added and `ZU.strToISO` removed.

**Comment by zuphilip** [comment]
Merged now! :rocket: Thank you very much @hubentu for all the work!

**Pull request merged**